### PR TITLE
VZ-8961: Build cluster-api v1.3.3 from source

### PR DESCRIPTION
**Pull request by Abhijithang**
This PR builds cluster-api v1.3.3 from source.

### CODE REVIEW

**Review by mgianatagh — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -33,39 +31,55 @@ ARG goproxy=https://proxy.golang.org
...
-RUN --mount=type=cache,target=/root/.cache/go-build \
-    --mount=type=cache,target=/go/pkg/mod \
-    CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} \
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} \
     go build -trimpath -ldflags "${ldflags} -extldflags '-static'" \
     -o manager ${package}
 
 # Production image
-FROM gcr.io/distroless/static:nonroot-${ARCH}
+FROM ghcr.io/oracle/oraclelinux:7-slim
```
Comment (bobdonat) [inline Dockerfile:71]: i think we are moving away from linux 7 and instead using linux 8.  linux 8 has a smaller footprint and supposedly less vulnerabilites.  I think @bhatpmk has a vz base image with ol8 that can be used here.

File: Dockerfile [hunk 2]
```diff
@@ -33,39 +31,55 @@ ARG goproxy=https://proxy.golang.org
...
+RUN go mod download
 
 # Copy the sources
 COPY ./ ./
 
-# Cache the go build into the Go’s compiler cache folder so we take benefits of compiler caching across docker build calls
-RUN --mount=type=cache,target=/root/.cache/go-build \
-    --mount=type=cache,target=/go/pkg/mod \
-    go build .
+## Cache the go build into the Go’s compiler cache folder so we take benefits of compiler caching across docker build calls
```
Comment (anders-swanson) [inline Dockerfile:55]: why is go build commented

File: cmd/clusterctl/Dockerfile [hunk 3]
```diff
@@ -33,37 +31,47 @@ ARG goproxy=https://proxy.golang.org
...
 # and so that source changes don't invalidate our downloaded layer
-RUN --mount=type=cache,target=/go/pkg/mod \
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} \
     go mod download
 
 # Copy the sources
 COPY ./ ./
 
 # Cache the go build into the Go’s compiler cache folder so we take benefits of compiler caching across docker build calls
-RUN --mount=type=cache,target=/root/.cache/go-build \
```
Comment (anders-swanson) [inline cmd/clusterctl/Dockerfile:49]: why are we removing the caching?

File: Dockerfile (hunk 2 above)
Comment (Abhijithang) [reply Dockerfile:55]: It was not intended. Thanks for finding it. Will fix it.

File: cmd/clusterctl/Dockerfile (hunk 3 above)
Comment (Abhijithang) [reply cmd/clusterctl/Dockerfile:49]: DOCKER_BUILDKIT  uses mount command for caching, so currently, we are not using this feature to build the image that requires a Docker server update in the VM agent image and requires extensive testing. Also, we use ephemeral VM instances, so dependencies are not stored anywhere to make the build faster.

File: Dockerfile (hunk 1 above)
Comment (Abhijithang) [reply Dockerfile:71]: Sure, I am checking with Bhat regarding using VZ base image/Oracle linux 8.

File: Dockerfile (hunk 2 above)
Comment (Abhijithang) [reply Dockerfile:55]: Added go build. Thanks.

**Pull request merged**